### DISCUSSION TITLE
Another Template Sensor

### DISCUSSION THREAD

**Discussion by porly1985** [JanM321/esphome-lg-controller#83] [General]
Hello everyone,

First of all, thank you for this amazing repository! It works perfectly!

I can confirm that the following indoor units are compatible:

- S09ET.NSJS
- S12ET.NSJS

With the recent option to switch from an internal to an external thermistor, I've created a template sensor to adjust the AC behavior. My primary goal was to minimize the ON/OFF cycles of the outdoor unit during heating and to maintain the inverter at a low power level.

```
   - trigger:
      - trigger: state
        entity_id: climate.lg_ac_og_local
        attribute: temperature
      - trigger: state
        entity_id: sensor.lumi_lumi_weather_temperature_5
      - trigger: state
        entity_id: input_number.ac_overheating_offset

    sensor:
      - name: "AC OG Controlled Room Temperature"
        unique_id: "ac_og_controlled_room_temperature"
        unit_of_measurement: "°C"
        device_class: temperature
        state_class: measurement
        state: >-
          {% set target_temp = state_attr('climate.lg_ac_og_local', 'temperature') | float(default=20) %}
          {% set overheating_offset = states('input_number.ac_overheating_offset') | float(default=0) %}
          {% set current_temp = states('sensor.lumi_lumi_weather_temperature_5') | float(default=20) %}
          {% set power_usage = states('sensor.shelly_plus_1pm_klima_switch_0_power') | float(default=0) %}
          {% set hvac_mode = states('climate.lg_ac_og_local') %}

          {% if trigger is defined and trigger.from_state is not none %}
            {% if trigger.entity_id == 'climate.lg_ac_og_local' and trigger.attribute == 'temperature' %}
              {% set prev_target_temp = trigger.from_state.attributes.temperature | float(default=20) %}
              {% set target_changed = True %}
            {% elif trigger.entity_id == 'sensor.lumi_lumi_weather_temperature_5' %}
              {% set target_changed = False %}
            {% else %}
              {% set target_changed = False %}
            {% endif %}
          {% else %}
            {% set target_changed = False %}
          {% endif %}

          {% if target_changed %}
            {{ current_temp }}
          {% elif hvac_mode == 'heat' %}
            {% if power_usage < 100 %}
              {{ current_temp }}
            {% elif current_temp >= target_temp + 3 + overheating_offset %}
              {{ current_temp }}
            {% elif current_temp >= target_temp + 2.5 %}
              {{ target_temp + 2.5 }}
            {% else %}
              {{ current_temp }}
            {% endif %}
          {% else %}
            {{ current_temp }}
          {% endif %}
```

I use this sensor with the default overheating setting set to 0. Heating starts when the target temperature +1°C is reached and stops when the target temperature +3°C is reached.

<img width="338" alt="image" src="https://github.com/user-attachments/assets/127a81d2-198d-4fff-bbd6-8e0d1b7737fd">

The main function of the template sensor is to "freeze" the temperature state just before the target temperature is reached. For instance, with a target temperature of 21°C, the sensor would hold its state at 23.5°C. With an overheating_offset set to 1, the sensor remains at 23.5°C until the room temperature exceeds 25°C. Because the inverter operates at a lower power level 0.5°C before reaching the target temperature, it's possible that the adjusted target temperature + overheating_offset will never be reached. This configuration allows the outdoor unit to run continuously at low power without cycling. I believe that reducing cycles can be more efficient and healthier for the AC unit.

<img width="632" alt="image" src="https://github.com/user-attachments/assets/1fa22931-a51e-44b7-82c6-a07b88ce79b1">

For my outdoorunit MU2R17 U12 I observed that defrosting starts exactly every 3 hours even if its 15 degrees outside and I dont know why... I would be happy to hear about your experiences.

**Comment by JanM321** [comment]
> First of all, thank you for this amazing repository! It works perfectly!

You're welcome! 🙂 

> For my outdoorunit MU2R17 U12 I observed that defrosting starts exactly every 3 hours even if its 15 degrees outside and I dont know why... I would be happy to hear about your experiences.

Unfortunately all LG Multi F units do this in heating mode (the single split units don't). It's called an 'oil return cycle'. There's some discussion of this in older issues.

It sucks also because after this happens, the unit tries to 'catch up' and uses a lot more power for a while. This means it's likely to reach its threshold and turn off.

I also have the MU2R17 outdoor unit and I've tried various things to control this better with the template sensor, but there's no *great* fix for it. That said, with the custom controller the behavior is pretty acceptable for me but I wouldn't buy this unit again.

**Comment by porly1985** [comment]
Great to hear that you have the same outdoor unit. If possible, I’d like to ask a question that isn’t related to your integration but rather about the AC itself. I’ve spent a lot of time searching for answers, but the LG community doesn’t have a strong presence on the internet.

<img width="416" alt="image" src="https://github.com/user-attachments/assets/c58adde3-1a28-4b67-8f55-ec7f6dd01e43">

<img width="349" alt="image" src="https://github.com/user-attachments/assets/5cc66e56-23b5-42e5-a6d4-fa33e75b42c2">

According to the user manual, I set the DIP switch to "Power Consumption Saving STEP 2" (with the AC disconnected from power). The max power consumption shouldn’t exceed 5.5 amps, which should be around 1300 watts.

<img width="408" alt="LG1" src="https://github.com/user-attachments/assets/e1b3727f-972c-4dde-b987-a931856df781">
 
When I turn on the AC (with an outdoor temperature of around 10°C), the inverter consumes more than 2000 watts, even with Power Save Mode STEP 2 active. When I set the target temperature to 30°C, it stays at more than 2000 watts, which I don’t think it should. After contacting LG Support twice, I finally received an answer with a new table. Now they say that Power Saving Mode Step 2 has a maximum of 7 amps and sides LG the issue is solved.

<img width="353" alt="image" src="https://github.com/user-attachments/assets/55f98972-da29-48fa-98db-fbb583e807f7">

Anyway, 7 amps would be around 1600 watts. I know my Shelly PM also measures both indoor units, but that wouldn’t explain a difference of  400W. I wonder if this behavior is normal... Have you experienced similar consumption values?

> I also have the MU2R17 outdoor unit and I've tried various things to control this better with the template sensor, but there's no great fix for it. That said, with the custom controller the behavior is pretty acceptable for me but I wouldn't buy this unit again.

Exactly! Without this integration, I would have quite a problem controlling my AC.

**Comment by JanM321** [comment]
> Anyway, 7 amps would be around 1600 watts. I know my Shelly PM also measures both indoor units, but that wouldn’t explain a difference of 400W. I wonder if this behavior is normal... Have you experienced similar consumption values?

I haven't done anything with the dip switches, but I vaguely remember this may only affect cooling mode? I don't know if that's true though. The LG units don't have a lot of advanced settings especially the multi-split units in heating mode.
